crates/core/src/network/net_util.rs

```diff
@@ -199,8 +201,15 @@ where
 
     tokio::select! {
         res = connection => {
-            // propagate errors
-            res.map_err(Error::HandlingConnection)?;
+            if let Err(e) = res {
+                if let Some(hyper_error) = e.downcast_ref::<hyper::Error>() {
+                    if hyper_error.is_incomplete_message() {
+                        debug!("Connection closed before request was complete");
```

Comment:
Not sure if the debug message is helpful without any context about the connection.

Reply:
Still figuring out how associated types work, now with added peer info!